### ISSUE TITLE
Playbook fails if it needs a package installed by pip during the playbook

### ISSUE THREAD

**Issue by orgito** [dw/mitogen#664]
- Ansible 2.9.1
- Mitogen 0.2.9

In one of my playbooks, I need to use the aws_s3 module. Since aws_s3 requires boto3 I have a task install it via pip before the aws_s3 task. If I use mitogen it will fail the first time. On a subsequent run it will work since the module is already present.

The playbook works flawlessly without mitogen.

**Comment by joshwaparadise** [comment]
I'm also having the same issue. 
ansible 2.8.4
mitogen-0.2.9
please let us know how this is to be taken care of

**Comment by ryanhope2** [comment]
I suspect that the "aggressive reuse of python processes" by mitogen has something to do with it.  Is there a way to make the re-use less aggressive under certain circumstances via a config or task?  maybe if we detect that boto was installed, we can execute a restart-python-processes type task

**Comment by orgito** [comment]
No. I simply stopped use mitogen for that playbook.

**Comment by schallee** [comment]
I'm seeing the same behavior installing docker-compose via pip and then using the ansible docker_compose module.

Not surprisingly, if I set mitogen_task_isolation to fork on the task using the docker_compose module it works. Doing that for every future usage isn't too practical but, oddly, it seems that setting mitogen_task_isolation to fork is not isolated to the current task. If I set it on the first use of the docker_compose module the subsequent use works without the setting.

So I guess a work around might be just to set fork on the next task.

**Comment by pun-ky** [comment]
yep, the second run usually is working. I am also blocked from using Mitogen because of this.
